controllers/opentelemetrycollectordistribution_controller_test.go

```diff
@@ -0,0 +1,347 @@
+	changed.Image = "quay.io/myns/my-dist:1.1.0"
+
+	err = k8sClient.Update(context.Background(), changed)
+	require.NoError(t, err)
+
+	req := k8sreconcile.Request{
+		NamespacedName: nsn,
+	}
+	_, err = reconciler.Reconcile(req)
+	require.NoError(t, err)
+
+	// verify
+	dist := cfg.Distribution(nsn.Namespace, nsn.Name)
+	assert.NotNil(t, dist)
+	assert.Equal(t, []string{"/path/to/command/changed"}, dist.Command)
+	assert.Equal(t, "quay.io/myns/my-dist:1.1.0", dist.Image)
+
+	// cleanup
+	require.NoError(t, k8sClient.Delete(context.Background(), dist))
+}
+
+func TestDeleteDistribution(t *testing.T) {
+	// prepare
+	nsn := types.NamespacedName{Name: "my-instance", Namespace: "default"}
+	cfg := config.New()
+	reconciler := NewDistributionReconciler(Params{
+		Client: k8sClient,
+		Log:    logger,
+		Scheme: testScheme,
+		Config: cfg,
+	})
+
+	{
+		existing := &v1alpha1.OpenTelemetryCollectorDistribution{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      nsn.Name,
+				Namespace: nsn.Namespace,
+			},
+			Command: []string{"/path/to/command"},
+			Image:   "quay.io/myns/my-dist:1.0.0",
+		}
+		err := k8sClient.Create(context.Background(), existing)
+		require.NoError(t, err)
+
+		req := k8sreconcile.Request{
+			NamespacedName: nsn,
+		}
+		_, err = reconciler.Reconcile(req)
+		require.NoError(t, err)
+
+		// sanity check
+		require.NotNil(t, cfg.Distribution(existing.Namespace, existing.Name))
+		require.Equal(t, []string{"/path/to/command"}, existing.Command)
+		require.Equal(t, "quay.io/myns/my-dist:1.0.0", existing.Image)
+	}
+
+	// test
+	deleted := &v1alpha1.OpenTelemetryCollectorDistribution{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      nsn.Name,
+			Namespace: nsn.Namespace,
+		},
+	}
+	err := k8sClient.Delete(context.Background(), deleted)
+	require.NoError(t, err)
+
+	req := k8sreconcile.Request{
+		NamespacedName: nsn,
+	}
+	_, err = reconciler.Reconcile(req)
+	require.NoError(t, err)
+
+	// verify
+	dist := cfg.Distribution(nsn.Namespace, nsn.Name)
+	assert.Nil(t, dist)
+}
+
+func TestCommandChanged(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		new      []string
+		old      []string
+		expected bool
+	}{
+		{
+			"unchanged",
+			[]string{"/path/to-command"},
+			[]string{"/path/to-command"},
+			false,
+		},
+		{
+			"item removed",
+			[]string{"/path/to-command"},
+			[]string{"/path/to-command", "second part"},
+			true,
+		},
+		{
+			"item added",
+			[]string{"/path/to-command", "second part"},
+			[]string{"/path/to-command"},
+			true,
+		},
+		{
+			"item changed",
+			[]string{"/path/to-command/updated"},
+			[]string{"/path/to-command"},
+			true,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.desc, func(t *testing.T) {
+			assert.Equal(t, tC.expected, commandChanged(tC.new, tC.old))
+		})
+	}
+}
+
+func TestRegisterDistributionControllerWithManager(t *testing.T) {
+	t.Skip("this test requires a real cluster, otherwise the GetConfigOrDie will die")
```

Comment:
Is it worth including this test then?

Reply:
If I have a minikube running, I can comment out this line and run this test manually. We might also want to start running minikube/kind in the CI, which would make this test work.